src/merge.rs

```diff
@@ -508,6 +607,7 @@ impl<'t, D: Driver, A: AbortSignal> Merger<'t, D, A> {
                 "Remote child {} already seen in another folder and merged",
                 remote_child_node
             );
+            merged_node.merge_state = merged_node.merge_state.with_new_remote_structure();
```

Comment:
Omitting a child that we already merged on the other side means we have new structure.

Reply:
This comment seems like it might as well be a, err, comment? :)